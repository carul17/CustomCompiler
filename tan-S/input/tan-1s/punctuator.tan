main {

    const pos := 'a' ; # true
    
    const b := 'a';
    var c := b;
    var d := 'a';
    var e := 'd';
    e := d;
    
    
    var s := "abc";
    var s2 := s;
    
    s := "bde";
    
    
    var s3 := "abc";
    var s4 := s2;
    
    var pos2 := 2 > 1; # true
    
    var pos3 := 2 < 1; #false
    
    var pos4 := 2 <= 1; #false
    
    var pos5 := 2 >= 1; # true
    
    var pos6 := 2 >= 2; # true
    
    var pos7 := 2 <= 2; # true
    
    var pos8 := 2 == 2; # true
    
    var pos9 := 2 == 3; #false
    
    var pos10 := 2 != 2; #false
    
    var pos11 := 2 != 3; # true
    
    var pos12 := true != true; # false
    
    var pos13 := true != false; # true
    
    var pos14 := true == true; # true
    
    var pos15 := true == false; # false
    
    
	print pos==b \n; #true
	print pos==c \n; #true
	print pos==d \n; #true
	print pos==e \n; #true
	print \n;
	print s == s2 \n; 
	print s2 \n;
	print s == s3 \n; 
	print s == s4 \n; 
	print s \n;
	print s4 \n;
	print \n;
	print pos2 \n;
	print pos3 \n;
	print \n;
	print pos4 \n;
	print pos5 \n;
	print pos6 \n;
	print pos7 \n;
	print \n;
	print pos8 \n;
	print pos9 \n;
	print \n;
	print pos10 \n;
	print pos11 \n;
	print \n;
	print pos12 \n;
	print pos13 \n;
	print pos14 \n;
	print pos15 \n;
}