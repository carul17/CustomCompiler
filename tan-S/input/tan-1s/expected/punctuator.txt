true
true
true
true

false
abc
false
false
bde
abc

true
false

false
true
true
true

true
false

false
true

false
true
true
false
